src/chain/chain.rs

```diff
@@ -1087,61 +1060,36 @@ impl Chain {
     ///     - if our name *is* the destination, returns an empty set; otherwise
     ///     - if the destination name is an entry in the routing table, returns it; otherwise
     ///     - returns the `N/3` closest members of the RT to the target
-    pub fn targets(
-        &self,
-        dst: &Authority<XorName>,
-        connected_peers: &[&XorName],
-    ) -> Result<BTreeSet<XorName>, Error> {
-        // FIXME: only filtering for now to match RT.
-        // should confirm if needed esp after msg_relay changes.
-        let is_connected = |target_name: &XorName| connected_peers.contains(&target_name);
-
+    pub fn targets(&self, dst: &Authority<XorName>) -> Result<(Vec<XorName>, usize), Error> {
```

Comment:
I am not clear why this function change so much as part of this PR.
Mostly what it seem we needed was the change in the return type that we have, and the last few line of the function to return that.

I am not saying it is not right, but this PR seem to combine both sending to peer we are not connected to, and doing the resend. Not sure it was necessary.

Reply:
Yes, this might be overextending the scope. I'll revert the changes related to `connected_peers`.